src/frontend/mame/ui/viewgfx.cpp

```diff
@@ -1093,13 +1093,14 @@ static void tilemap_handler(mame_ui_manager &mui, render_container &container, u
 			ypixel = (mapboxheight - 1) - ypixel;
 		if (state.tilemap.rotate & ORIENTATION_SWAP_XY)
 			std::swap(xpixel, ypixel);
-		int col = ((xpixel / pixelscale + state.tilemap.xoffs) / tilemap->tilewidth()) % tilemap->cols();
-		int row = ((ypixel / pixelscale + state.tilemap.yoffs) / tilemap->tileheight()) % tilemap->rows();
-		int gfxnum, code, color;
+		UINT32 col = ((xpixel / pixelscale + state.tilemap.xoffs) / tilemap->tilewidth()) % tilemap->cols();
+		UINT32 row = ((ypixel / pixelscale + state.tilemap.yoffs) / tilemap->tileheight()) % tilemap->rows();
+		UINT8 gfxnum;
+		UINT32 code, color;
 		tilemap->get_info_debug(col, row, gfxnum, code, color);
 		util::stream_format(title_buf, " @ %d,%d = GFX%d #%X:%X",
 							col * tilemap->tilewidth(), row * tilemap->tileheight(),
-							gfxnum, code, color);
+							int(gfxnum), code, color);
```

Comment:
Minor thing, util::stream_format is very forgiving on the types you pass for the format string - to print a `UINT8` in decimal, you can just use `%u` and not bother cast it.

The one thing it does that can be surprising from a `printf` PoV is that `%u`/`%d` apply no promotion to short/char types before formatting.  For example `printf("%d", UINT8(130))` yields `130` because the `UINT8` is first promoted to `unsigned` when calling the varargs function before being cast to `int` when formatting.  On the other hand, `util::stream_format(str, "%d", UINT8(130))` yields `-126` because no promotion occurs, so it casts `UINT8` to `INT8` and prints it.